wit/proxy.wit

```diff
@@ -0,0 +1,45 @@
+// The `wasi:http/proxy` world captures a widely-implementable intersection of
+// hosts that includes HTTP forward and reverse proxies. Components targeting
+// this world may concurrently stream in and out any number of incoming and
+// outgoing HTTP requests.
+default world proxy {
+  // HTTP proxies have access to time and randomness.
+  import random: random.random
+  // TODO: add `import wall-clock: clocks.wall-clock`
+  // TODO: add `import monotonic-clock: clocks.monotonic-clock`
+
+  // This is the default logging handler to use when user code simply wants to
+  // log to a developer-facing console (e.g., via `console.log()`).
+  import console: logging.handler
+
+  // TODO: Once the underlying Wit template machinery is implemented, add:
+  //
+  //   import loggers: interface {
+  //     *: logging.handler
+  //   }
+  //
+  // which will allow a component to import any number of non-default logging
+  // backends that different categories of log messages can be sent to.
+
+  // TODO: add `import metrics: metrics.counters`
+
+  // This is the default handler to use when user code simply wants to make an
+  // HTTP request (e.g., via `fetch()`) but doesn't otherwise specify a
+  // particular handler.
+  import default-upstream-HTTP: pkg.outgoing-handler
```

Comment:
HTTP world is already overloaded with terms regarding direction, with `upstream`/`downstream` often meaning opposite things in different documents and software. Could we avoid using multiple terms describing the same thing in this spec? Maybe let's stick to `upstream`/`downstream`, so this should be changed to:
`import default-upstream-HTTP: pkg.upstream-handler`

Reply:
Yep, my mistake mixing and then inverting terminologies.  [Fixed](https://github.com/WebAssembly/wasi-http/pull/3/commits/afec79da5b4d8afec7610dccb77fc4a8cb21b0b8)